src/main/java/core/basesyntax/DataConverterImpl.java

```
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+import core.interfaces.DataConverter;
+
+import java.io.UncheckedIOException;
+import java.util.Arrays;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        return data.stream()
+                .skip(1)
+                .filter(s -> !s.isEmpty())
+                .map(this::toTransaction)
+                .toList();
+    }
+
+    private FruitTransaction toTransaction(String data) {
+        String[] split = data.split(",");
+        if (split.length != 3) { throw new IllegalArgumentException("Bad line" + Arrays.toString(split)); }
+        FruitTransaction.Operation operation = FruitTransaction.Operation.fromCode(split[0]);

```

Operation mapping uses `FruitTransaction.Operation.fromCode(split[0])` which will throw for unknown codes, but you should `trim()` the code before calling and wrap exceptions to provide an informative error message that includes the input line (checklist: "Operation code maps to a known Operation; otherwise fail fast with informative error"). Provide a clear message with line content and file context when possible.